src/eventhandler/mod.rs

```diff
@@ -0,0 +1,85 @@
+//! Helper to run a callback in the libuv main thread.
+
+use std::mem;
+use std::os::raw::c_void;
+
+use types::*;
+use handle::{Handle, Managed};
+use context::*;
+use neon_runtime;
+use neon_runtime::raw;
+use std::sync::Arc;
+
+struct EventHandlerInner(*mut c_void);
+
+unsafe impl Send for EventHandlerInner {}
+unsafe impl Sync for EventHandlerInner {}
+
+impl Drop for EventHandlerInner {
+    fn drop(&mut self) {
+        unsafe {
+            neon_runtime::threadsafecb::delete(self.0);
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct EventHandler(Arc<EventHandlerInner>);
+
+impl EventHandler {
+    pub fn new(callback: Handle<JsFunction>) -> Self {
+        TaskContext::with(|mut cx: TaskContext| {
+            let cb = unsafe {
```

Comment:
To reduce the amount of `unsafe`, could this method delegate to `bind` instead?

```rust
pub fn new(callback: Handle<JsFunction>) -> Self {
    TaskContext::with(|mut cx| Self::bind(cx.global(), callback))
}
```

Reply:
Absolutely, thanks for pointing out.